src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -266,6 +315,54 @@ private ManifestVersion GetInstalledManifestVersion(ManifestId manifestId)
             }
         }
 
+        private bool AdManifestSentinalIsDueForUpdate()
+        {
+            var sentinalPath = GetAdvertisingManifestSentinalPath();
+            int updateIntervalHours;
+            if (!int.TryParse(Environment.GetEnvironmentVariable("DOTNET_CLI_WORKLOAD_UPDATE_NOTIFY_INTERVAL_HOURS"), out updateIntervalHours))
+            {
+                updateIntervalHours = 24;
+            }
+
+            if (File.Exists(sentinalPath))
+            {
+                var lastAccessTime = File.GetLastAccessTime(sentinalPath);
+                if (lastAccessTime.AddHours(updateIntervalHours) > DateTime.Now)
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private async Task<bool> UpdatedAdManifestPackagesExistAsync()
+        {
+            var manifests = GetInstalledManifestIds();
+            var avaliableUpdates = await Task.WhenAll(manifests.Select(manifest => NewerManifestPackageExists(manifest)))
+                .ConfigureAwait(false);
+            return avaliableUpdates.Any();
+        }
+
+        private async Task<bool> NewerManifestPackageExists(ManifestId manifest)
+        {
+            try
+            {
+                var currentVersion = NuGetVersion.Parse(_workloadResolver.GetManifestVersion(manifest.ToString()));
+                var latestVersion = await _nugetPackageDownloader.GetLatestPackageVerion(GetManifestPackageId(_sdkFeatureBand, manifest));
+                return latestVersion > currentVersion;
+            }
+            catch (Exception)
+            {
+                return false;
+            }
+        }
+
+        private bool BackgroundUpdatesAreDisabled() => 
+            bool.TryParse(Environment.GetEnvironmentVariable("DOTNET_CLI_WORKLOAD_UPDATE_NOTIFY_DISABLE"), out var disableEnvVar) && disableEnvVar;
```

Comment:
there is an existing method GetEnvironmentVariableAsBool in src/Cli/Microsoft.DotNet.Cli.Utils/Env.cs